Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt

```diff
@@ -106,14 +111,9 @@ class ConfidenceFeatureProvider private constructor(
         oldContext: EvaluationContext?,
         newContext: EvaluationContext
     ) {
-        if (newContext != oldContext) {
-            // on the new context we want to fetch new values and update
-            // the storage & cache right away which is why we pass `InitialisationStrategy.FetchAndActivate`
-            internalInitialize(
-                newContext,
-                InitialisationStrategy.FetchAndActivate
-            )
-        }
+        val context = newContext.toConfidenceContext()
+        val removedKeys = oldContext?.asMap()?.keys?.minus(newContext.asMap().keys) ?: emptySet()
+        confidence.putContext(context.map, removedKeys.toList())
```

Comment:
Just a thought: could we updated `removedKeys` from here and just call `confidence.putContext(context.map)`? `newContext` should already not include any key that is in the `removedKeys` list

Reply:
I don't think we can: `putContext` just puts this keys in the confidence context map, if we had a, b, c and now new context has d, if we put d we are going to have a, b, c, d. consider that putContext doesn't replace the whole context.